webapp/repos.py

```diff
@@ -48,6 +48,21 @@ def _filter_modified_since(self, repos_to_process, modified_since_dt):
                     break
         return filtered_repos_to_process
 
+    def _filter_third_party(self, repos_to_process, third_party):
+        filtered_repos = []
+
+        if third_party:
+            return repos_to_process
+
+        for label in repos_to_process:
+            for repo_id in self.cache.repolabel2ids.get(label, []):
+                repo_detail = self.cache.repo_detail[repo_id]
+                # If we don't want third party repo, and this repo is flagged as third party, skip it
+                if not third_party and repo_detail[REPO_THIRD_PARTY]:
```

Comment:
You don't need to check "third_party" again here. It's surely False.

Reply:
Well, maybe not, but it expresses the intent more clearly.

Reply:
I don't think so. It's just confusing.